### PR TITLE
Add index selector docs

### PR DESCRIPTION
**Pull request by garrensmith**
This is added for https://github.com/apache/couchdb/pull/808

## Checklist

- [ ] Documentation is written and is accurate;
- [ ] `make check` passes with no errors
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the commit hash once this PR is rebased and merged

### CODE REVIEW

File: src/api/database/find.rst [hunk 1]
```diff
@@ -1007,6 +1009,50 @@ Example index creation using all available query parameters
       "skip": 0
     }
 
+Creating an index with a Selector
+=================================
+
+Adding a selector to the index will add more fine grained filtering
+of what documents will be added to an index. At query time,
+the index must be specified via the use_index field for the
```
Comment (willholley) [inline src/api/database/find.rst]: perhaps link to use_index (or _find) documentation

File: src/api/database/find.rst [hunk 2]
```diff
@@ -1007,6 +1009,50 @@ Example index creation using all available query parameters
       "skip": 0
     }
 
+Creating an index with a Selector
+=================================
+
+Adding a selector to the index will add more fine grained filtering
```
Comment (flimzy) [inline src/api/database/find.rst]: <grammar nazi>fine-grained should be hyphenated</grammar nazi>

File: src/api/database/find.rst [hunk 3]
```diff
@@ -1007,6 +1009,50 @@ Example index creation using all available query parameters
...
+of what documents will be added to an index. At query time,
+the index must be specified via the use_index field for the
+query planner to use it.
+
+Example index with a selector, only documents that contain
+the `age` and `sport` fields and that match the selector will be
+added to the index.
```
Comment (willholley) [inline src/api/database/find.rst]: I think worth mentioning that the query selector is independent from the indexed fields. E.g. in this example, you could have a selector on `{"type": "player")`, for example. This really comes into play when you want to use a partial index to restrict an index in a way that cannot be done at query time. For example, by filtering on `$ne` or `$regex`.

**Comment by garrensmith** [comment]
Replaced by https://github.com/apache/couchdb-documentation/pull/163

**Pull request closed without merging**